cmd/query/main.go

```diff
@@ -12,7 +12,7 @@ import (
 	swaggerFiles "github.com/swaggo/files"
 	ginSwagger "github.com/swaggo/gin-swagger"
 
-	"github.com/equinor/vds-slice/api"
+	"github.com/equinor/vds-slice/controllers"
```

Comment:
Several places. Some examples:
https://golang.withcodeexample.com/blog/chapter-3-project-structure-in-gin-framework/
https://github.com/vsouza/go-gin-boilerplate
https://medium.com/@wahyubagus1910/build-scalable-restful-api-with-golang-gin-gonic-framework-43793c730d10

The old oneseismic repo contained code that was not related to the user facing API such as definition of the data format and a system for distributed data fetching and extraction. Now the entire repo is the API.

Reply:
OK, makes sense. However:

Controllers in your examples are parts of bigger model: mvc or some controller/service/router concept.
We do not structure our code as any of them.
(And I would have at least understood if you named this directory handlers (as at least word "handle" is a major part of the directory you are dealing with), but controllers??? New concept!)

If you want to follow any of these structures properly, you apparently need to split cmd into cmd and router, move all endpoint methods to services. Or split stuff into server/models/something, depending on which link I should consult.

I don't think it is that necessary though, as basically every project has their own structure, doesn't look like there is one everyone follows.

So consider if you want to confirm to some standard because you find current structure unreadable. Then it is better to discuss the proper structure in the team and then do one big change.

But if you only want to lightly change understanding to "everything is just an api", then at least use `handlers` instead of controllers :smiling_imp:

Reply:
My goal was not to follow these project structures. I noticed that the code was hard to read since the code for handling each type of request was split in ways that did not make sense to me. I chose the name controllers as it seemed to be the most common, but I am okay with calling this folder handlers or even endpoints.

Reply:
The important thing here is that I think that it makes sense to split code according to endpoints. That is how our API is structured, and we chose that structure for a reason right?

Reply:
The API is not structured to always divide code into fence/slice etc files.
It depended on file size and on logical likeliness.

For example datahandle deals with all requests in one file, because it being datahandle is more important than splitting traces from samples.
capi.cpp also has all the calls together because the purpose is to turn c requests into c++ requests, doesn't matter what these requests are.
If all requests definitions were in the same file it would also have been fine. But they are not coupled that much that I think that they must stay together.
etc

So if you lean into major refactoring I would recommend thinking about what your final solution (after all the future refactoring you were talking about) would look like and discussing it in the team beforehand.

If you want to do just this one part of the job, api -> handlers or endpoints is fine and putting everything defined on SliceRequest class and etc in their own files is fine too.

Reply:
The API, as in the (web) interface that we publish is structured that way. And this code is responsible defining that API, so splitting it up by endpoints makes sense I think.